tools/benchmarks/classification/knn_imagenet/test_knn.py

```diff
@@ -145,30 +139,41 @@ def main():
     else:
         logger.info('No pretrained or checkpoint is given, use random init.')
 
-    if not distributed:
-        model = MMDataParallel(model, device_ids=[0])
-    else:
+    if torch.cuda.is_available():
+        model = model.cuda()
+
+    if distributed:
         model = MMDistributedDataParallel(
-            model.cuda(),
+            module=model.cuda(),
             device_ids=[torch.cuda.current_device()],
             broadcast_buffers=False)
 
-    model.eval()
-    # build extraction processor and run
-    extractor = ExtractProcess()
-    train_feats = extractor.extract(
-        model, data_loader_train, distributed=distributed)['feat']
-    val_feats = extractor.extract(
-        model, data_loader_val, distributed=distributed)['feat']
+    if is_model_wrapper(model):
+        model = model.module
+
+    # build extractor and extract features
+    extractor_train = Extractor(
+        extract_dataloader=data_loader_train,
+        seed=args.seed,
+        dist_mode=distributed,
+        pool_cfg=copy.deepcopy(dataset_cfg.pool_cfg))
+    extractor_val = Extractor(
+        extract_dataloader=data_loader_val,
+        seed=args.seed,
+        dist_mode=distributed,
+        pool_cfg=copy.deepcopy(dataset_cfg.pool_cfg))
+    train_feats = extractor_train(model)['feat5']
+    val_feats = extractor_val(model)['feat5']
```

Comment:
Is it better not to write hard code here?

Reply:
We only need the last stage avgpooled feature here.